torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,513 @@
+
+            if modal not in os.listdir(root) and len(self.modality_urls[modal]) == 2:
+                self.merge_parts(root, modal)  # same modality two times
+
+        if "mask" in self.modality:
+            self.modality.remove("mask")
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        self.list_modals = []
+        id = self.ids[index]
+
+        mask = self._load_raster(id, "mask")
+        mask[mask != 0] = 1
+
+        if "s2_temporal_subset" in self.modality:
+            s2_temporal_subset = self._load_raster(id, "s2_temporal_subset")
+            s2_temporal_subset_index = torch.index_select(
+                s2_temporal_subset, dim=0, index=self.band_indices
+            )
+            self.list_modals.append(s2_temporal_subset_index)
+
+        if "s2_autumn" in self.modality:
+            s2_autumn = self._load_raster(id, "s2_autumn")
+            s2_autumn_index = torch.index_select(
+                s2_autumn, dim=0, index=self.band_indices
+            )
+            self.list_modals.append(s2_autumn_index)
+
+        if "s2_spring" in self.modality:
+            s2_spring = self._load_raster(id, "s2_spring")
+            s2_spring_index = torch.index_select(
+                s2_spring, dim=0, index=self.band_indices
+            )
+            self.list_modals.append(s2_spring_index)
+
+        if "s2_summer" in self.modality:
+            s2_summer = self._load_raster(id, "s2_summer")
+            s2_summer_index = torch.index_select(
+                s2_summer, dim=0, index=self.band_indices
+            )
+            self.list_modals.append(s2_summer_index)
+
+        if "s2_winter" in self.modality:
+            s2_winter = self._load_raster(id, "s2_winter")
+            s2_winter_index = torch.index_select(
+                s2_winter, dim=0, index=self.band_indices
+            )
+            self.list_modals.append(s2_winter_index)
+
+        if "esa_wc" in self.modality:
+            esa_wc = self._load_raster(id, "ESA_WC")
+            self.list_modals.append(esa_wc)
+
+        if "viirs" in self.modality:
+            viirs = self._load_raster(id, "VIIRS")
+            self.list_modals.append(viirs)
+
+        if "s1" in self.modality:
+            s1 = self._load_raster(id, "s1")
+            self.list_modals.append(s1)
+
+        image = torch.cat(self.list_modals, dim=0)
+
+        if self.transforms is not None:
+            im = np.einsum("ijk->jki", image.numpy())
+            msk = np.einsum("ijk->jki", mask.numpy())
+            sample_: dict[str, Tensor] = {"image": im, "mask": msk}
+
+            transformed = self.transforms(sample_)
+
+            image = torch.Tensor(np.einsum("ijk->kij", transformed["image"]))
+            mask = torch.Tensor(np.einsum("ijk->kij", transformed["mask"]))
```

Comment:
Preferably all tensors would be in the shape expected by our transforms without reshaping. Also, no need to convert to numpy and back to PyTorch just to reshape, you can reshape directly in PyTorch using einops.

Reply:
Good point -somehow missed this.